pdr_backend/sim/sim_engine.py

```diff
@@ -171,16 +173,38 @@ def run_one_iter(self, test_i: int, mergedohlcv_df: pl.DataFrame):
         acct_up_profit -= stake_up
         acct_down_profit -= stake_down
 
-        # trader: enter the trading position
-        usdcoin_holdings_before = st.holdings[self.usdcoin]
-        if pred_up:  # buy; exit later by selling
+        if pred_up and self.position_open == "":
+            # Open long position if pred up and no position open
             usdcoin_amt_send = trade_amt * conf_up
-            tokcoin_amt_recd = self._buy(curprice, usdcoin_amt_send)
-        elif pred_down:  # sell; exit later by buying
-            target_usdcoin_amt_recd = trade_amt * conf_down
-            p = self.ppss.trader_ss.fee_percent
-            tokcoin_amt_send = target_usdcoin_amt_recd / curprice / (1 - p)
-            self._sell(curprice, tokcoin_amt_send)
+            tok_received = self._buy(curprice, usdcoin_amt_send)
+            self.position_open = "long"
+            self.position_worth = usdcoin_amt_send
+            self.position_size = tok_received
+            st.trader_profits_USD.append(0)
+        elif pred_down and self.position_open == "":
+            # Open short position if pred down and no position open
+            tokcoin_amt_send = trade_amt * conf_down / curprice
+            usd_received = self._sell(curprice, tokcoin_amt_send)
+            self.position_open = "short"
+            self.position_worth = usd_received
+            self.position_size = tokcoin_amt_send
+            st.trader_profits_USD.append(0)
+        elif self.position_open == "long" and not pred_up:
```

Comment:
The large # if statements is hard to track. A couple suggestions:
- group by if position_open = "long" vs "short". Could be nested within, though better is one case vs the other.
- better than "not pred_down" and "not pred_up": perhaps you have three states: "pred_down", "pred_nomove", "pred_up"

Reply:
Good Q. Try both, see what comes out cleanest. The overall if/then/else code block should look clean.

---
A related idea: maybe the if/then/else can go into its own method? (If not too many args)